add-dynamic-entity-layer/src/main/java/com/esri/arcgismaps/sample/adddynamicentitylayer/components/MapViewModel.kt

```diff
@@ -0,0 +1,136 @@
+/* Copyright 2023 Esri
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.esri.arcgismaps.sample.adddynamicentitylayer.components
+
+import android.app.Application
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.AndroidViewModel
+import com.arcgismaps.mapping.ArcGISMap
+import com.arcgismaps.mapping.BasemapStyle
+import com.arcgismaps.mapping.Viewpoint
+import com.arcgismaps.mapping.layers.DynamicEntityLayer
+import com.arcgismaps.realtime.ArcGISStreamService
+import com.arcgismaps.realtime.ArcGISStreamServiceFilter
+import com.esri.arcgismaps.sample.adddynamicentitylayer.R
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class MapViewModel(
+    application: Application,
+    private val sampleCoroutineScope: CoroutineScope,
+) : AndroidViewModel(application) {
+
+    // set the state of the switches and slider
+    val trackLineCheckedState = mutableStateOf(false)
+    val prevObservationCheckedState = mutableStateOf(false)
+    val trackSliderValue = mutableStateOf(5f)
+
+    // Flag to show or dismiss the bottom sheet
+    val isBottomSheetVisible = mutableStateOf(false)
+
+    // set the MapView mutable stateflow
+    val mapViewState = MutableStateFlow(MapViewState())
+
+    // create ArcGIS Stream Service
+    private val streamService =
+        ArcGISStreamService(application.getString(R.string.stream_server_url))
+
+    // create ArcGISStreamServiceFilter
+    private val streamServiceFilter = ArcGISStreamServiceFilter()
+
+    // layer displaying the dynamic entities on the map
+    private val dynamicEntityLayer: DynamicEntityLayer
+
+    /**
+     * set the data source for the dynamic entity layer.
+     */
+    init {
+        // set condition on the ArcGISStreamServiceFilter to limit the amount of data coming from the server
+        streamServiceFilter.apply {
+            whereClause = "speed > 0"
+        }
```

Comment:
Unnecessary apply block used here
```suggestion
        streamServiceFilter.whereClause = "speed > 0"
```